### PR TITLE
PWN-4616 Restore onboarding flow on pin creation

### PR DESCRIPTION
**Pull request by kuznetsov**
### Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-4616

### Description of Work

Navigate to pin code creation if user left onboarding on this step

### CODE REVIEW

**Review by kabramovp2p — APPROVED** [review]

**Review by 1sakov — CHANGES_REQUESTED** [review]

**Review by 1sakov — APPROVED** [review]

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/auth/ui/onboarding/root/OnboardingRootPresenter.kt [hunk 1]
```diff
@@ -1,26 +1,34 @@
...
+            userSignUpDetailsStorage.getLastSignUpUserDetails() != null -> view.navigateToContinueOnboarding()
+            else -> view.navigateToOnboarding()
         }
 
         // Sign in unidentified user for help messenger in onboarding flow
         IntercomService.signIn(emptyString())
     }
+
+    private fun userLeftOnPinCreation(): Boolean {
+        return tokenKeyProvider.secretKey.isNotEmpty() && !authInteractor.isAuthorized()
```
[the remark this answers is not in the document]
Comment (kuznetsov) [reply app/src/main/java/org/p2p/wallet/auth/ui/onboarding/root/OnboardingRootPresenter.kt:32]: secretKey instance guarantees us the user got all 3 shares, cause account is creating after otp confirmation

File: app/src/main/java/org/p2p/wallet/auth/ui/onboarding/root/OnboardingRootFragment.kt [hunk 2]
```diff
@@ -22,20 +24,21 @@ class OnboardingRootFragment :
...
+
+    override fun navigateToCreatePin() {
+        navigateTo(NewCreatePinFragment.create())
+    }
+
+    private fun navigateTo(fragment: Fragment) = replaceFragment(
+        target = fragment,
+        containerId = R.id.onboardingRootContainer,
+        fragmentManager = childFragmentManager,
+        addToBackStack = false
```
[the remark this answers is not in the document]
Comment (kuznetsov) [reply app/src/main/java/org/p2p/wallet/auth/ui/onboarding/root/OnboardingRootFragment.kt:42]: @kabramovp2p @gslevinkov guys could you help me with that?

File: app/src/main/java/org/p2p/wallet/auth/ui/onboarding/root/OnboardingRootPresenter.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/auth/ui/onboarding/root/OnboardingRootPresenter.kt:32]: ah, got it!

File: app/src/main/java/org/p2p/wallet/auth/ui/onboarding/root/OnboardingRootFragment.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/auth/ui/onboarding/root/OnboardingRootFragment.kt:42]: you have to call `popAndReplace` on the next screen.
for example: 

`previous screen` - `OnboardingRootFragment` - `next screen` - `target screen`

if you want the user to navigate from `next screen` to `target screen` and by clicking on back on `target screen` you want the user to be on `previous screen` you should call: 

popAndReplace(target = `target screen`, popTo=`previous screen`)

In that case you'll skip the `OnboardingRootFragment` without manipulations with backstack

[the remark this answers is not in the document]
Comment (kuznetsov) [reply app/src/main/java/org/p2p/wallet/auth/ui/onboarding/root/OnboardingRootFragment.kt:42]: I've got it. We can consider changing onboarding navigation

**Pull request merged**